x-pack/plugins/apm/server/lib/correlations/get_correlations_for_slow_transactions/get_latency_distribution.ts

```diff
@@ -107,7 +107,14 @@ export async function getLatencyDistribution({
 
   function formatDistribution(distribution: Agg['distribution']) {
     const total = distribution.doc_count;
-    return distribution.dist_filtered_by_latency.buckets.map((bucket) => ({
+
+    // remove trailing buckets that are empty and out of bounds of the desired number of buckets
+    const buckets = dropRightWhile(
+      distribution.dist_filtered_by_latency.buckets,
+      (bucket, index) => bucket.doc_count === 0 && index > intervalBuckets - 1
+    );
```

Comment:
I don't understand why we are doing this. If we want fewer bucket returned we shouldn't we specify a larger interval?

Reply:
Btw. all this should matter less if we use a logarithmic scale

Reply:
I created a separate issue to address it https://github.com/elastic/kibana/issues/91572

Reply:
I created a separate issue for the distribution scale: https://github.com/elastic/kibana/issues/91574